src/UserModule/service/user.service.ts

```diff
@@ -72,12 +72,21 @@ export class UserService {
     const role: Role = await this.roleService.findByRoleName(user.role);
     const salt: string = this.createSalt();
     const hashPassword: string = this.createHashedPassword(user.password, salt);
-    return this.repository.save({
-      ...user,
-      salt,
-      password: hashPassword,
-      role,
-    });
+    let savedUser: User;
+    try {
+      savedUser = await this.repository.save({
```

Comment:
```suggestion
      return await this.repository.save({
```
Pode retornar direto, mas deixa o await para cair no catch de falhar